src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -840,7 +841,7 @@ bool ObjectMonitor::deflate_monitor(Thread* current) {
     install_displaced_markword_in_object(obj);
   }
 
-  if (event.should_commit()) {
+  if (obj != nullptr && event.should_commit()) {
```

Comment:
So you decided to just drop the event completely ... but is that what people tracking these events would want? Do we need to update anything on the event definition so people know this is the case?

Reply:
With regards to the event definition @egahlin suggests:
> It might be worth mentioning it in the field description, e.g. "If null or N/A, the object has been garbage collected", but that is all. (Null is represented as 0 in the file format, which the parser will interpret as a missing value. Both 'jfr print' and JMC will display "N/A")

Reply:
That is if null is used instead of dropping the event. I don't have an opinion on what is the best approach as it depends on how the event is being used.

Reply:
I don't have a strong preference for either dropping event or emitting it with N/A values. I get the argument for replacing `_sync_Deflations` to count the actual deflations, so lets emit the event with N/A values then. Amended in new commit.

Unfortunately, I see no reliable way to test exactly the case for emitting the N/A event: we need to sequence monitor deflation and GC precisely to get to that point reliably. So the targeted event test just tests the always-reachable case, while the new stress test covers the corner case of dead object, among other things.